test/IISExpress.FunctionalTests/HttpsTests.cs

```diff
@@ -117,39 +120,76 @@ private async Task HttpsHelloWorldCerts(TestVariant variant, bool sendClientCert
             }
         }
 
+        private X509Certificate2 CreateRootCert()
+        {
+            var parentKey = CreateKeyMaterial(2048);
+
+            // On first run of the test, creates the certificate in the trusted root certificate authorties.
+            var parentRequest = new CertificateRequest("CN=IISIntegrationTest_Root", parentKey, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+
+            parentRequest.CertificateExtensions.Add(
+                new X509BasicConstraintsExtension(
+                    certificateAuthority: true,
+                    hasPathLengthConstraint: false,
+                    pathLengthConstraint: 0,
+                    critical: true));
+
+            parentRequest.CertificateExtensions.Add(
+                new X509KeyUsageExtension(X509KeyUsageFlags.DigitalSignature | X509KeyUsageFlags.NonRepudiation, critical: true));
+
+            parentRequest.CertificateExtensions.Add(
+                new X509SubjectKeyIdentifierExtension(parentRequest.PublicKey, false));
+
+            var notBefore = DateTimeOffset.Now.AddDays(-1);
+            var notAfter = DateTimeOffset.Now.AddYears(5);
+
+            var parentCert = parentRequest.CreateSelfSigned(notBefore, notAfter);
+
+            // Need to export/import the certificate to associate the private key with the cert.
+            var imported = parentCert;
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                // On non OSX systems we need to export the certificate and import it so that the transient
+                // key that we generated gets persisted.
+                var export = parentCert.Export(X509ContentType.Pkcs12, "");
+                imported = new X509Certificate2(export, "", X509KeyStorageFlags.PersistKeySet | X509KeyStorageFlags.Exportable);
+                Array.Clear(export, 0, export.Length);
+            }
+
+            // Add the cert to the cert store
+            using (var store = new X509Store(StoreName.Root, StoreLocation.LocalMachine))
```

Comment:
Admin only?

This kind of state change shouldn't be made on dev machines.

Reply:
We might be better off pre-provisioning a test root cert on the CI agents.

Reply:
Added a check for running as admin.

Reply:
Adding a trusted cert is not something that should be automated if a 3rd party runs build.cmd on this repo. @blowdart

Reply:
We can add a check for running on only VSTS, jenkins, and appveyor.

Reply:
A fixed test root would be better. Spin up a Win Server CA and issue the certs from there, import the root CA via powershell (this is what I do for the cert auth). This could also exercise CRL checking.